### PR TITLE
Feature/set price on listing

### PR DESCRIPTION
**Pull request by boivlad**
Task -https://app.clickup.com/t/860pd6x2q

### CODE REVIEW

File: frontend/app/features/assets/models/investAssetModel.ts [hunk 1]
```diff
@@ -94,13 +98,28 @@ const $doValidateFormValues = atom(null, (get, set, args: UiAssetInvestFormChang
...
   const form = get($form)
-  if (form.isValid) {
-    await router.push(paths.investAsset({id}))
+  if (!form.isValid) {
+    return;
+  }
+  const provider = get($userRpcProvider) as ethers.providers.JsonRpcProvider
+  const usdtTokenSigned = get($usdtSmartContractSigned) as Contract
+  const smartContract = new ethers.Contract(scAddress, tokenABI, provider);
+  const assetSc = smartContract.connect(provider.getSigner()) as Contract
```
Comment (ruslanpalagin) [inline frontend/app/features/assets/models/investAssetModel.ts:113]: we should have a model for this

File: frontend/app/features/assets/views/AssetInvest/index.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ type InvestBlockProps = {
   tokensTotalSupply: BigNumber,
   irr: number,
   coc: number,
-  id: number,
+  scAddress: string,
```
Comment (ruslanpalagin) [inline frontend/app/features/assets/views/AssetInvest/index.tsx:16]: No need to pass is down to send than it up.... Just use scAddress in controller

File: contracts_arbitrum/contracts/v3/Asset.sol [hunk 3]
```diff
@@ -80,4 +81,9 @@ contract Asset is IAsset, ERC20 {
       rewardsDistributor.onUserBalanceChanged(recipient, recipientBalance);
     }
   }
+
+  function setStatus(Status _status) external override {
+    require(msg.sender == addressesProvider.getSeriesMasterAdmin(), "only seriesMasterAdmin can set status");
```
[the remark this answers is not in the document]
Comment (boivlad) [reply contracts_arbitrum/contracts/v3/Asset.sol:86]: Agree with u

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx [hunk 4]
```diff
@@ -71,74 +73,100 @@ export const AssetsDetailsController = (): JSX.Element => {
...
-        <div className={s.assetInvest}>
-          <div className={clsx(s.assetInvestBuy, s.container)}>
-            <AssetInvest {...investInfo} balance={balance} account={account} form={form} onChangeForm={onChangeForm}
-                         onFormSubmit={onFormSubmit}/>
+        <div className={ s.assetInvest }>
+          <div className={ clsx(s.assetInvestBuy, s.container) }>
+            <AssetInvest { ...investInfo } scAddress={ $asset.scAddress } balance={ balance }
+                         account={ account } form={ form }
+                         onChangeForm={ onChangeForm }
+                         onFormSubmit={ async ({ scAddress }: { scAddress: string }) => {
```
[the remark this answers is not in the document]
Comment (boivlad) [reply frontend/app/features/assets/controllers/AssetsDetailsController.tsx:159]: We receive scAddress from atom

File: frontend/app/features/assets/models/investAssetModel.ts [hunk 5]
```diff
@@ -94,13 +98,28 @@ const $doValidateFormValues = atom(null, (get, set, args: UiAssetInvestFormChang
...
-  const id = get($assetId)
   const form = get($form)
-  if (form.isValid) {
-    await router.push(paths.investAsset({id}))
+  if (!form.isValid) {
+    return;
+  }
+  const provider = get($userRpcProvider) as ethers.providers.JsonRpcProvider
+  const usdtTokenSigned = get($usdtSmartContractSigned) as Contract
+  const smartContract = new ethers.Contract(scAddress, tokenABI, provider);
```
[the remark this answers is not in the document]
Comment (boivlad) [reply frontend/app/features/assets/models/investAssetModel.ts:112]: Agree, will rename to readable name(for ex.: tokenAddress)

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ruslanpalagin) [reply frontend/app/features/assets/controllers/AssetsDetailsController.tsx:159]: Yes, no need to get from view. Because we have it already

**Review by ruslanpalagin — APPROVED** [review]

**Pull request merged**